TypeCobol/Compiler/Scanner/Token.cs

```diff
@@ -21,24 +21,23 @@ public class Token : Antlr4.Runtime.IToken, IVisitable
 
         public override bool Equals(object obj)
         {
-            if (!(obj is Token))
-                return false;
-            var tokenCompare = (Token) obj;
-
-            return this.Type == tokenCompare.Type
-                   && this.Channel == tokenCompare.Channel
-                   && this.Column == tokenCompare.Column
-                   && this.EndColumn == tokenCompare.EndColumn
-                   && this.ExpectedClosingDelimiter == tokenCompare.ExpectedClosingDelimiter
-                   && this.Length == tokenCompare.Length
-                   && this.Line == tokenCompare.Line
-                   && this.StartIndex == tokenCompare.StartIndex
-                   && this.StopIndex == tokenCompare.StopIndex
-                   && this.TokenIndex == tokenCompare.TokenIndex
-                   && this.HasClosingDelimiter == tokenCompare.HasClosingDelimiter
-                   && this.LiteralValue == tokenCompare.LiteralValue
-                   && this.TokenType == tokenCompare.TokenType
-                   && this.Text == tokenCompare.Text;
+            if (obj is Token token)
+            {
+                return token.Type == this.Type &&
+                       token.StartIndex == this.StartIndex &&
+                       token.StopIndex == this.StopIndex &&
+                       token.TokensLine == this.TokensLine;
```

Comment:
I tried to use TokenSource in the beginning but I had trouble since this property may or may not be set. Two distinct tokens may both have a null `TokenSource` and comparing them will yield wrong result.
It appears that `Type`, `StartIndex`, `StopIndex` and `TokensLine` are always set and are constitutive to the Token. A new `TokenLine` is created in `CompilationDocument` (creation or update) so two tokens defined one in a COPY, the other in a Program will have different TokensLine and will be considered distinct.

Reply:
Can you test this case:
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. RB1LEQU1.
       data division.
       working-storage section.
       COPY YTOTO
       01 toto pic X.
      
       procedure division.
           move toto to toto
           .
       end program RB1LEQU1.
```
And COPY YTOTO contains:
     01 toto pic 9.
at line 6.

Reply:
Actually the token in the copy should be at line 9 to match the one in the move statement because the comparison is made on userFilterToken which is positioned where the hover is requested.
Here are my findings in debug mode :
- the SymbolInformationForTokens dictionary does not include tokens from COPY so it won't create any conflict
- using the Watcher in VS, I managed to compare two similar tokens written one in a COPY, the other in main program : the two tokens are considered distinct by the Equals method and they differ only in their TokensLine which is the expected behavior.

Also, the hover request on a variable defined in a COPY won't return anything.